OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -120,12 +120,14 @@ public override bool HandleMouseInput(MouseInput mi)
 							}
 						}
 					}
-					else if (dragStart.HasValue)
+					else if (dragStart.HasValue && (!(World.OrderGenerator is GenericSelectTarget) || hasBox))
```

Comment:
I removed the `dragStart.HasValue` part now, it indeed didn't make a difference. Also found a way to place the comment in such way that it pleases StyleCop.